tiledb/sm/c_api/tiledb.cc

```diff
@@ -2954,6 +2954,26 @@ int32_t tiledb_query_add_range(
   return TILEDB_OK;
 }
 
+int32_t tiledb_query_add_range_by_name(
+    tiledb_ctx_t* ctx,
+    tiledb_query_t* query,
+    const char* dim_name,
+    const void* start,
+    const void* end,
+    const void* stride) {
+  if (sanity_check(ctx) == TILEDB_ERR || sanity_check(ctx, query) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  uint32_t dim_idx;
+  SAVE_ERROR_CATCH(ctx, query->query_->get_dimension_index(dim_name, &dim_idx));
+
+  if (SAVE_ERROR_CATCH(
+          ctx, query->query_->add_range(dim_idx, start, end, stride)))
+    return TILEDB_ERR;
```

Comment:
This definitely works, but we try to keep the C API as thin as possible -- even if that means bloating the internal interfaces a little bit.

Instead of performing the name-to-idx conversion with `get_dimension_index`, let's introduce a `tiledb::sm::Query::add_range_by_name()` routine and call that directly.

Reply:
Agreed, I followed this suggestion in my new patch.